libraries/chain/include/eosio/chain/wasm_interface_collection.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+#include <eosio/chain/wasm_interface.hpp>
+
+namespace eosio::chain {
+
+   /**
+    * @class wasm_interface_collection manages the active wasm_interface to use for execution.
+    */
+   class wasm_interface_collection {
+      public:
+         wasm_interface_collection(wasm_interface::vm_type vm, wasm_interface::vm_oc_enable eosvmoc_tierup,
+                                   const chainbase::database& d, const std::filesystem::path& data_dir,
+                                   const eosvmoc::config& eosvmoc_config, bool profile)
+            : main_thread_id(std::this_thread::get_id())
+            , wasm_runtime(vm)
+            , eosvmoc_tierup(eosvmoc_tierup)
+            , wasmif(vm, eosvmoc_tierup, d, data_dir, eosvmoc_config, profile)
+         {}
+
+         wasm_interface& get_wasm_interface() {
+            if (is_on_main_thread() || is_eos_vm_oc_enabled()) {
+               return wasmif;
+            }
+            return *threaded_wasmifs[std::this_thread::get_id()];
+         }
+
+
+         // update current lib of all wasm interfaces
+         void current_lib(const uint32_t lib) {
+            // producer_plugin has already asserted irreversible_block signal is called in write window
+            wasmif.current_lib(lib);
+            for (auto& w: threaded_wasmifs) {
+               w.second->current_lib(lib);
+            }
+         }
+
+         // only called from non-main threads (read-only trx execution threads) when producer_plugin starts them
+         void init_thread_local_data(const chainbase::database& d, const std::filesystem::path& data_dir,
+                                     const eosvmoc::config& eosvmoc_config, bool profile) {
+            EOS_ASSERT(!is_on_main_thread(), misc_exception, "init_thread_local_data called on the main thread");
+            if (is_eos_vm_oc_enabled()) {
+               // EOSVMOC needs further initialization of its thread local data
+               wasmif.init_thread_local_data();
```

Comment:
lost an `#ifdef EOSIO_EOS_VM_OC_RUNTIME_ENABLED` on these couple lines during the refactor

Reply:
I wonder if most of the `#ifdef EOSIO_EOS_VM_OC_RUNTIME_ENABLED` could be removed to simplify differences. For now, I'll just revert back to having them.

Reply:
Restored `#ifdef`s